app/controllers/onboarding_controller.rb

```diff
@@ -24,6 +27,31 @@ def create
 
   def success; end
 
+  # rubocop:disable Metrics/AbcSize
+  def telegram_auth
+    check_hash = telegram_auth_params[:hash]
+    auth_data = telegram_auth_params.slice(:auth_date, :first_name, :id, :last_name)
+    check_string = auth_data.to_unsafe_h.map { |k, v| "#{k}=#{v}" }.sort.join("\n")
+
+    secret_key = OpenSSL::Digest::SHA256.new.digest(ENV['TELEGRAM_BOT_API_KEY'])
+    hash = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'), secret_key, check_string)
```

Comment:
Does it work similar like JWT? Would it be ok if we write a test for that method and throw in an example token?